src/D2L.CodeStyle.Analyzers/Diagnostics.cs

```diff
@@ -485,5 +485,15 @@ public static class Diagnostics {
 			description: "[ReadOnly] paramaters must not be assigned to or passed by non-readonly reference."
 		);
 
+		public static readonly DiagnosticDescriptor DontCallDefaultStructConstructor = new DiagnosticDescriptor(
+			id: "D2LXXXX",
+			title: "Don't call default struct constructor",
+			messageFormat: "Don't call default struct constructor",
+			description: "The default parameterless constructor for a struct is overridable, but not removable. It is quite possible it's not doing what you expect. Try another constructor or initializer?",
+			category: "Correctness",
+			defaultSeverity: DiagnosticSeverity.Info,

```

Comment:
Info because probably want to actually think about changes.

Reply:
And I'll have to review the instances to see if this is generally true (that the default constructor is undesirable)